/*
 * This file was automatically generated by EvoSuite
 * Fri Apr 07 17:02:50 KST 2017
 */

package org.prevayler.foundation.network;

import static org.evosuite.runtime.EvoAssertions.assertThrownBy;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.fail;

import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(useVNET = true) 
public class NetworkReceiverFactory_ESTest3 extends NetworkReceiverFactory_ESTest_scaffolding3 {

  @Test
  public void test0()  throws Throwable  {
      StubbornNetworkImpl stubbornNetworkImpl0 = new StubbornNetworkImpl();
      StubbornServerReceiverImpl stubbornServerReceiverImpl0 = (StubbornServerReceiverImpl)stubbornNetworkImpl0.newReceiver((Service) null, (ObjectSocket) null);
      assertNotNull(stubbornServerReceiverImpl0);
  }

  @Test
  public void test1()  throws Throwable  {
      NetworkImpl networkImpl0 = new NetworkImpl();
      // Undeclared exception!
      try { 
        networkImpl0.newReceiver((Service) null, (ObjectSocket) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("org.prevayler.foundation.network.NetworkClientObjectReceiverImpl", e);
      }
  }
}
